generators/uscore-r4/generator.rb

```diff
@@ -306,6 +300,44 @@ def get_search_params(search_parameters, sequence)
   search_code
 end
 
+def get_comparator_searches(search_params, sequence)
+  search_code = ''
+  search_assignments = []
+  search_params.each do |param|
+    search_assignments << "'#{param}': #{param_value_name(param)}"
+  end
+  search_assignments_str = "{ #{search_assignments.join(', ')} }"
+  search_params.each do |param|
+    param_val_name = param_value_name(param)
+    param_info = sequence[:search_param_descriptions][param.to_sym]
+    comparators = param_info[:comparators]
+    type = param_info[:type]
+    case type
+    when 'Period', 'date'
```

Comment:
I decided to do a case in case we need to add other types in the future. Right now only period and date types are required by US Core.

Reply:
I would definitely vote against adding useless code to the repo because it might become useful later. It is just as possible that when we need to add more types we determine that a different approach is necessary and then this code never becomes useful.

Reply:
I don't know. Using case matches more closely with the appropriate underlying logic to me. I mean the alternative is to just do a 'next unless' which doesn't really match what I had in mind for the function. Like yeah, it's technically not necessary to do a case, but I think it fits better aesthetically, and it's only adding one line anyways. Anyways, if you feel strongly about it, I'm willing to just go with whatever.

Reply:
I think it's a good idea for us to avoid placeholder code as well.  Also, if another type with a comparator suddenly shows up and the code in here isn't generic enough to handle it, then we should raise an exception.  So could you get rid of the `case` and put in the following guard clause (or something like it), if it makes sense to you?

```
raise 'Only Period and date types for search comparators supported' unless ['Period','date'].include? type